internal/internal_task_handlers.go

```diff
@@ -1005,19 +1022,33 @@ ProcessEvents:
 		}
 		reorderedEvents := nextTask.events
 		markers := nextTask.markers
-		historyMessages := nextTask.msgs
+		historyMessages := nextTask.acceptedMsgs
 		flags := nextTask.flags
 		binaryChecksum := nextTask.binaryChecksum
 		nextTaskBuildId := nextTask.buildID
+		admittedUpdates := nextTask.admittedMsgs
 		// Check if we are replaying so we know if we should use the messages in the WFT or the history
 		isReplay := len(reorderedEvents) > 0 && reorderedHistory.IsReplayEvent(reorderedEvents[len(reorderedEvents)-1])
 		var msgs *eventMsgIndex
 		if isReplay {
+			requestUpdateBodies := make(map[string]*anypb.Any)
+			for _, updateRequest := range admittedUpdates {
+				requestUpdateBodies[updateRequest.GetProtocolInstanceId()] = updateRequest.GetBody()
+			}
+			// Check if we need to replace the body of the update request with the body of the update admitted event
+			for _, msg := range historyMessages {
+				if body, ok := requestUpdateBodies[msg.GetProtocolInstanceId()]; ok {
+					msg.Body = body
```

Comment:
To confirm, you want to replace this even if body is `nil` on the admitted request? Or I guess we can assume body is never nil in an update request?

Reply:
Yeah I am assuming the latter, @dandavison the admitted body would never be `nil` correct?